test/spec/functions/merge.js

```diff
@@ -0,0 +1,244 @@
+'use strict';
+const merge = require('../../../src/functions/merge');
+
+it('should merge `source` into `object`', function() {
+  var names = {
+    characters: [{name: 'barney'}, {name: 'fred'}]
+  };
+
+  var ages = {
+    characters: [{age: 36}, {age: 40}]
+  };
+
+  var heights = {
+    characters: [{height: '5\'4"'}, {height: '5\'5"'}]
+  };
+
+  var expected = {
+    characters: [
+      {name: 'barney', age: 36, height: '5\'4"'},
+      {name: 'fred', age: 40, height: '5\'5"'}
+    ]
+  };
+
+  expect(merge(names, ages, heights)).toEqual(expected);
+});
+
+it.skip('should merge sources containing circular references', function() {
+  var object = {
+    foo: {a: 1},
+    bar: {a: 2}
+  };
+
+  var source = {
+    foo: {b: {c: {d: {}}}},
+    bar: {}
+  };
+
+  source.foo.b.c.d = source;
+  source.bar.b = source.foo.b;
+
+  var actual = merge(object, source);
+
+  expect(actual.bar.b).not.toEqual(actual.foo.b);
+  expect(actual.foo.b.c.d).toEqual(actual.foo.b.c.d.foo.b.c.d);
+});
+
+it('should work with four arguments', function() {
+  var expected = {a: 4};
+  var actual = merge({a: 1}, {a: 2}, {a: 3}, expected);
+
+  expect(actual).toEqual(expected);
+});
+
+it('should merge onto function `object` values', function() {
+  function Foo() {}
+
+  var source = {a: 1};
+  var actual = merge(Foo, source);
+
+  expect(actual).toEqual(Foo);
+  expect(Foo.a).toEqual(1);
+});
+
+it('should merge first source object properties to function', function() {
+  var fn = function() {};
+  var object = {prop: {}};
+  var actual = merge({prop: fn}, object);
```

Comment:
Is it the expected behaviour? I would have thought that the object replaces the function.

```js
_.merge({ props() {} }, { props: {} }) // { props: {} } 
```

Reply:
not if the target is a function apparently, this test is the same as in lodash

Reply:
_.merge({ props() {} }, { props: {} })

Reply:
lodash behaves differently if a function is top-level or as a value of the object, we don't. Since we don't have functions in the first place, this isn't breaking any usages.